### PR TITLE
feat: currencyManager manages conversion paths

### PR DESCRIPTION
**Pull request by benjlevesque**
The idea is to move the aggregator paths out of the library, so that services can automatically pull the values from the smart contracts

Next step: expose the utility to pull from smart contracts

Also in this PR: generalize the use of `UnsupportedCurrencyError`

### CODE REVIEW

File: packages/advanced-logic/src/extensions/payment-network/any-to-erc20-proxy.ts [hunk 1]
```diff
@@ -1,18 +1,19 @@
+import { conversionSupportedNetworks, ICurrencyManager } from '@requestnetwork/currency';
 import { ExtensionTypes, RequestLogicTypes } from '@requestnetwork/types';
 import Erc20FeeProxyPaymentNetwork from './erc20/fee-proxy-contract';
-import { currenciesWithConversionOracles } from './conversion-supported-currencies';
 
 const CURRENT_VERSION = '0.1.0';
 
 export default class AnyToErc20ProxyPaymentNetwork extends Erc20FeeProxyPaymentNetwork {
   public constructor(
+    private currencyManager: ICurrencyManager,
```
Comment (kevindavee) [inline packages/advanced-logic/src/extensions/payment-network/any-to-erc20-proxy.ts:9]: Any reason why currencyManager must be injected?

**Review by alexandre-abrioux — APPROVED** [review]

File: packages/currency/src/errors.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { RequestLogicTypes } from '@requestnetwork/types';
+
+const getSymbolAndNetwork = (
+  symbolWithNetwork: string | Omit<RequestLogicTypes.ICurrency, 'type'>,
+) => {
+  if (typeof symbolWithNetwork === 'string') {
+    const [symbol, network] = symbolWithNetwork.split('-');
```
Comment (yomarion) [inline packages/currency/src/errors.ts:7]: (unrelated to the PR) This does not work well if the symbol contains a dash. I think we already handled this somewhere else.

File: packages/currency/test/chainlink-path-aggregators.test.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      [USD.hash]: {
+        [fakeDAI.hash]: 1,
+        [EUR.hash]: 1,
+      },
+      [EUR.hash]: {
+        [USD.hash]: 1,
+      },
+    },
+  };
+  it('can get a path between 2 currencies with no aggregator', () => {
```
Comment (yomarion) [inline packages/currency/test/chainlink-path-aggregators.test.ts:25]: I think it's worth it testing that unexisting paths are not found

**Review by yomarion — APPROVED** [review]
Perfect !

**Review by KolevDarko — APPROVED** [review]

**Pull request merged**